File: contains-duplicate-ii/contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        ios_base::sync_with_stdio(false); cin.tie(NULL);
        unordered_map<int, int> umap;
        for(int i=0; i<nums.size(); ++i) {
            if (umap.count(nums[i]) > 0 && abs(umap[nums[i]] - i) <= k) 
                return true;
            umap[nums[i]] = i;
        }
        return false;
    }
};

File: contains-duplicate-ii/contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> umap = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            if (umap.containsKey(nums[i]) && Math.abs(umap.get(nums[i]) - i) <= k) {
                return true;
            }
            umap.put(nums[i], i);
        }
        return false;
    }
}
